FirebaseMessaging/Sources/FIRMessagingContextManagerService.m

```diff
@@ -194,6 +166,42 @@ + (void)scheduleiOS10LocalNotificationForMessage:(NSDictionary *)message atDate:
   }
 }
 
++ (UNMutableNotificationContent *)contentFromContextualMessage:(NSDictionary *)message {
+  UNMutableNotificationContent *content = [[UNMutableNotificationContent alloc] init];
+  NSDictionary *apsDictionary = message;
+
+  // Badge is universal
+  if (apsDictionary[kFIRMessagingContextManagerBadgeKey]) {
+    content.badge = apsDictionary[kFIRMessagingContextManagerBadgeKey];
+  }
+#if TARGET_OS_IOS || TARGET_OS_OSX || TARGET_OS_WATCH
+  // The following fields are not available on tvOS
+  if ([apsDictionary[kFIRMessagingContextManagerBodyKey] length]) {
+    content.body = apsDictionary[kFIRMessagingContextManagerBodyKey];
+  }
+  if ([apsDictionary[kFIRMessagingContextManagerTitleKey] length]) {
+    content.title = apsDictionary[kFIRMessagingContextManagerTitleKey];
+  }
+
+  if (apsDictionary[kFIRMessagingContextManagerSoundKey]) {
+#if !TARGET_OS_WATCH
+    content.sound =
+        [UNNotificationSound soundNamed:apsDictionary[kFIRMessagingContextManagerSoundKey]];
+#endif
```

Comment:
nit: Could you please add comments marking which it belongs to for readability here and at other similar conditions, e.g. 
```
#endif // !TARGET_OS_WATCH
```

Reply:
Done.